### ISSUE TITLE
Add ability to filter sub-arrays

### ISSUE THREAD

**Issue by rick-nu** [particle-php/Filter#7]
Add the ability to filter sub-arrays

```
$f = new Particle\Filter;

$f->value('user.first_name')->trim();
$f->value('user.last_name')->trim();
$f->value('username')->trim();

$f->filter([
    'username' => ' Rick ',
    'user' => [
        'first_name' => ' Rick ',
        'last_name' => ' van der Staaij ',
    ],
]);
```

**Comment by rick-nu** [comment]
Fixed in https://github.com/particle-php/Filter/pull/21

**Comment by rick-nu** [comment]
Released `v0.4.0` with sub-array filter support. Closing issue.
